pkg/controllers/clusterregistration/controller.go

```diff
@@ -98,7 +99,7 @@ func saToClusterRegistration(namespace, name string, obj runtime.Object) ([]rela
 	if sa, ok := obj.(*v1.ServiceAccount); ok {
 		ns := sa.Annotations[fleet.ClusterRegistrationNamespaceAnnotation]
 		name := sa.Annotations[fleet.ClusterRegistrationAnnotation]
-		if ns != "" && name != "" && len(sa.Secrets) > 0 {
+		if ns != "" && name != "" {
```

Comment:
About this change: There is no way to grab the token of the SA here since no ctx is passed to use the cache/api lookup. But as long as the SA has non-empty ClusterRegistrationAnnotation, it should be ok to enqueue that SA to be checked by ClusterRegistration controllers.